File: implementation/lib/func.ml
type t = {
  name   : string;
  params : string list;
  body   : Stmt.t list;
}

let create_store (func : t) (vals : Val.t list) : Store.t =
  let varvals = List.combine func.params vals in
  Store.create varvals


let  print_list (lis:string list) : string =
	String.concat ", "  lis


let create (name : string) (params : string list) (body : Stmt.t list) : t = { name; params; body }

let get_name (func : t) : string = func.name

let get_params (func : t) : string list = func.params

let get_body (func : t) : Stmt.t list = func.body

let str (func : t) : string =  "\nfunction " ^ func.name ^ " (" ^ (print_list func.params)^") {\n"^"Oi"^"\n}"

File: JS2ECMA-SL/src/ECMA-SL/syntax/Func.js
const Stmt = require("./Stmt/Stmt");

class Func {
  constructor(name, params, body) {
    this.name = name;
    this.params = params;
    this.body = body;
  }

  toString() {
    return `function ${this.name} (${this.params}) {\n${this.body.toString()}\n}`;
  }
}

Func.fromJSON = function (obj) { 
  
  var name = obj.name; 
  var params = obj.params;
  var body = Stmt.fromJSON(obj.body);  
  return new Func(name, params, body); 
}

module.exports = Func;
